application/src/main/java/org/togetherjava/tjbot/features/basic/CoolMessagesBoardManager.java

```diff
@@ -0,0 +1,129 @@
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.CoolMessagesBoardConfig;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.awt.*;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * Manager for the cool messages board. It appends highly-voted text messages to a separate channel
+ * where members of the guild can see a list of all of them.
+ */
+public final class CoolMessagesBoardManager extends MessageReceiverAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(CoolMessagesBoardManager.class);
+    private static final Emoji REACT_EMOJI = Emoji.fromUnicode("🌟");
+    private final Predicate<String> boardChannelNamePredicate;
+    private final CoolMessagesBoardConfig config;
+
+    public CoolMessagesBoardManager(Config config) {
+        this.config = config.getCoolMessagesConfig();
+
+        boardChannelNamePredicate =
+                Pattern.compile(this.config.boardChannelPattern()).asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReactionAdd(MessageReactionAddEvent event) {
+        final MessageReaction messageReaction = event.getReaction();
+        int originalReactionsCount = messageReaction.hasCount() ? messageReaction.getCount() : 0;
+        boolean isCoolEmoji = messageReaction.getEmoji().getName().equals(REACT_EMOJI.getName());
+        long guildId = event.getGuild().getIdLong();
+        Optional<TextChannel> boardChannel = getBoardChannel(event.getJDA(), guildId);
+
+        if (boardChannel.isEmpty()) {
+            logger.warn("Tried to find the board channel in server {} but was unable to do so",
+                    guildId);
+            return;
+        }
+
+        // If the bot has already reacted to this message, then this means that
+        // the message has been quoted to the cool messages board, so skip it.
+        if (hasBotReacted(event.getJDA(), messageReaction)) {
+            return;
+        }
+
+        if (isCoolEmoji && originalReactionsCount + 1 >= config.minimumReactions()) {
+            event.retrieveMessage().queue(message -> {
+                message.addReaction(REACT_EMOJI).queue();
+
+                insertCoolMessage(boardChannel.orElseThrow(), message);
+            }, e -> logger.warn("Tried to retrieve cool message but got: {}", e.getMessage()));
+        }
+    }
+
+    /**
+     * Gets the board text channel where the quotes go to, wrapped in an optional.
+     *
+     * @param jda the JDA
+     * @param guildId the guild ID
+     * @return the board text channel
+     */
+    private Optional<TextChannel> getBoardChannel(JDA jda, long guildId) {
+        return jda.getGuildById(guildId)
```

Comment:
```java
jda
 .getGuildById(guildId)
 .getTextChannelsByName(channel.getName(), true)
.stream()
.findAny();```
Wouldn't this be better?

Reply:
We could definitely change it! Although I believe it wouldn't change a lot of things :)

Reply:
What does `getTextChannelCache()` do? If it's using a cache, does it clear up? How is this cache loaded?

Reply:
Okay, I read the docs, I think as long as the channel exists before the bot is started, it should be okay.

Reply:
> What does `getTextChannelCache()` do? If it's using a cache, does it clear up? How is this cache loaded?

the jda starts it loads all guilds and channels in cache